ballerina/jsonpath.bal

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2024 WSO2 LLC (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/jballerina.java;
+import ballerina/lang.'object as obj;
+
+public type JsonPathValue ()|boolean|int|float|decimal|string|json[]|map<json>;
```

Comment:
Why can't we use `json` type instead of defining a new type?

Reply:
Did it for improve the understandability of the code.
But I think json will be better, will change it